src/DensityTest.jl

```diff
@@ -0,0 +1,272 @@
+# Density test for manipulation in the running variable. 
+# This work is based on McCrary, J. (2008). Manipulation of the running variable in the regression discontinuity design: A density test. Journal of econometrics, 142(2), 698-714.
+
+"""
+    McCraryTest(bin::Union{Real,Nothing} = nothing, bw::Union{Real,Nothing} = nothing) 
+
+A McCrary (2008) test for manipulation in the running variable for `SharpRD` estimator.
+
+The `bin` object is the width of each bin. The default is 2*sd(runvar)*length(runvar)^(-0.5), following McCrary 2008.
+
+The `bw` object is the bandwidth. The default uses the calculation from McCrary (2008), pg. 705.
+"""
+Base.@kwdef struct McCraryTest 
+    bin::Union{Real,Nothing} = nothing 
+    bw::Union{Real,Nothing} = nothing 
+end
+
+Base.@kwdef struct FittedMcCraryTest{T,M,V,C}
+    b::T
+    h::T
+    θhat::T
+    σθ::T
+    pval::T
+    Js::M
+    fitted_res::V
+    coeftable::C
+end
+
+
+# 0. Kernel used in the McCrary Test
+
+TriangularKernel(t) = max(0, (1 - abs(t)))
```

Comment:
I think this does not necessarily need to be defined. You could use `pdf(SymTriangularDist(), t)` instead.  In fact perhaps in the main `fit` function you could define, (with h the bandwidth)

`kernel = SymTriangularDist(0, h)`

and then write `pdf(kernel, x)` instead of `TriangularKernel(x/h)/h`.

This would perhaps [NOTE: comment below may well be outside the scope of this PR :) just brainstorming] also allow to extend this function to use more general kernels (see `kernels.jl` and `local_linear.jl` for an example) by making `kernel` a field of `McCraryTest`. If I see it correctly, the only point for which you use the fact that you are specifically using the triangular kernel is that  κ = 3.348. Do you know what the formula is for κ that would allow its computation for other kernels?

Reply:
You are right, it is better to make `kernel` a field of `McCraryTest`. I did that. However, as you notice, this brings problems with κ. In McCrary (2008) he mentions that "[...] The standard formula (for κ) (see Fan and Gijbels, 1996, Eq. (4.3)) does not apply to the boundary case (see Fan and Gijbels, Eqs (3.20) and (3.22)). The cited constant is specific to the triangle kernel in the boundary case." (footnote 18). I don't know how to calculate that constant. From the cited footnote, it is not clear for me where to find the formula. 

I created a new method to the `kernel_constant' function, but it only gives you the constant for the triangular kernel (κ = 3.348). The `bandwidth` function takes this function, so once we realize how to create that constant for a family of kernels, it should work for the package!

Reply:
Sounds great, thanks!